javascript/src/main/components/Statistics/SingleCourseSectionForm.js

```diff
@@ -0,0 +1,66 @@
+import React, { useState } from "react";
+import { Form, Button, Spinner } from "react-bootstrap";
+import DepartmentFormSelect from "main/components/Statistics/DepartmentFormSelect";
+import QuarterFormSelect from "main/components/Statistics/QuarterFormSelect";
+
+const SingleCourseSectionForm = ({ setCourseJSON, fetchJSON, onSubmit = () => {} }) => {
+    const [startQuarter, setStartQuarter] = useState("20204");
+    const [endQuarter, setEndQuarter] = useState("20211");
+    const [department, setDepartment] = useState("CMPSC");
+    const [loading, setLoading] = useState(false);
+    const [courseNumber, setCourseNumber] = useState("");
+    const [courseSuf, setCourseSuf] = useState("");
+
+    const handleSubmit = (event) => {
+        event.preventDefault();
+        onSubmit();
+        setLoading(true);
+        fetchJSON({startQuarter, endQuarter, department, courseNumber, courseSuf})
+        .then((courseJSON)=> {
+            setCourseJSON(courseJSON);
+            setLoading(false);
+        });
+    };
+
+    const handleCourseNumberOnChange = (event) => {
+        const rawCourse = event.target.value;
+        if (rawCourse.match(/\d+/g) != null) {
+            const number = rawCourse.match(/\d+/g)[0];
+            setCourseNumber(number);
+        } 
+        
+        if (rawCourse.match(/[a-zA-Z]+/g) != null) {
+            const suffix = rawCourse.match(/[a-zA-Z]+/g)[0];
+            setCourseSuf(suffix);
+        } else {
+            setCourseSuf("");
+        }
+    };
```

Comment:
This is duplicated code from CourseSearchCourseStartEndQtr.js (lines35-50), this code should probably be refactored out into a utility file. This is fine for this PR, but if you all are up for it, you could refactor this out from both places to cut down on duplicated code.